From c0a84393b8f959551e991fc8fc0608d89aa9cac9 Mon Sep 17 00:00:00 2001
From: Alona Solntseva <alona.solntseva@tandemg.com>
Date: Wed, 16 Feb 2022 11:43:30 +0200
Subject: [PATCH] [SW-67906] gaudi2: erom pcie: return initial dbi

Save initial pci aux dbi value and write it back at the end of
pcie_config_iatu function

Signed-off-by: Alona Solntseva <alona.solntseva@tandemg.com>
Change-Id: I038ebfd73d2a4e77587e3401b8ddd5e12646fc02
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 350c4d8..c32cc72 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -552,6 +552,7 @@
 void pcie_config_iatu(void)
 {
 	uint32_t pci_dbi_base = (mmPCIE_DBI_BASE & 0x7FFFFFF);
+	uint32_t pci_aux_dbi = regmap_readl(mmPCIE_AUX_DBI);
 
 	regmap_writel(DBI_ENABLE, mmPCIE_AUX_DBI);
 	regmap_writel(IATU_LOWER_TARGET_ADDR,
@@ -562,6 +563,7 @@
 		      (pci_dbi_base + IATU_REGION_CTRL_1_OFF_INBOUND_OFF));
 	regmap_writel(IATU_INBOUND_HI_OFFSET_ADDR,
 		      (pci_dbi_base + IATU_REGION_CTRL_2_OFF_INBOUND_OFF));
+	regmap_writel(pci_aux_dbi, mmPCIE_AUX_DBI);
 }
 
 void enable_flr_handling_fsm(void)
